File: leetcode/problems_7_reverse-integer.go
package main

import (
	"fmt"
)

func reverse (x int) int {
	var negative = false
	var origin int64
	if x < 0 {
		origin = -int64(x)
		negative = true
	} else {
		origin = int64(x)
	}

	var reverted int64 = 0
	var digit int64
	
	for origin > 0 {
		origin, digit = origin / 10, origin % 10
		reverted = reverted * 10 + digit
	}

	if negative {
		reverted = -reverted
	}

	if reverted < -2147483648 || reverted > 2147483647 {
		reverted = 0
	}

	return int(reverted)
}

func main () {
	fmt.Println(reverse(-111111111111111111))
}

File: leetcodecn/problems_7_reverse-integer.java
class Solution {
    public int reverse(int x) {
	int y = 0;
	while (x >= 10 || x <= -10) {
	    y = y * 10 + x % 10;
	    x /= 10;
	}
	if (x != 0 && ((y * 10) / 10 != y)) {
	    return 0;
	}
	return y * 10 + x % 10;
    }
}
